zulipterminal/ui_tools/boxes.py

```diff
@@ -125,6 +123,21 @@ def stream_box_view(self, stream_id: int, caption: str='', title: str='',
             (self.msg_write_box, self.options()),
         ]
         self.contents = write_box
+        self.set_stream_write_box_caption(self.contents[0][0][0].edit_text)
+
+    def set_stream_write_box_caption(self, stream_name: Optional[str]) -> None:
```

Comment:
can you write a test for this? your code coverage has decreased because of this.

Reply:
Sure. I'll update the tests once that function is properly refactored.
Thanks for having a look at it :)